### ISSUE TITLE
Ability to comment on saved insights

### ISSUE THREAD

**Issue by corywatilo** [PostHog/posthog#10320]
## Is your feature request related to a problem?

**TL;DR: I need clarification on a saved insight.**

When I have a product usage question, I start by searching saved insights. But when I see the insight I'm looking for was last updated 8 months ago, I get a sneaking suspicion it's out of date... But I'd like confirmation before changing an insight I didn't create.

**The problem:** There's no natural place to have conversations about specific data _inside_ PostHog... which means I have to revert to Slack.

If I move to Slack, the answer is likely to get lost into the abyss - and that's assuming the question isn't asked in a DM. And the answer will likely be useful for others. (See where this is going? * cough * Squeak!)

## Describe the solution you'd like

Before making changes to a saved insight that someone else created, I'd like to gut check my thought. Basically I want comments for saved insights

[Here's a 4-page prototype](https://balsamiq.cloud/sd0i9zq/pl0wjty/r5CA7?f=N4IgUiBcAMA0IDkoEZ4GEXwDKZAISjhAFlcAtQ%2BAaSgG0BdAXyA%3D) demoing a commenting experience for saved insights, with the most interesting wireframe (final step) below:

![image](https://user-images.githubusercontent.com/154479/173987300-7aec9f3e-49ab-4a16-8bca-dca0d35b4a0c.png)

## Describe alternatives you've considered

Slack. Email. In-person meetup. Just not asking and biasing for editing.

**Comment by clarkus** [comment]
Cool idea. I'd want to think about this in the context of other things that can be created or edited. Comments are really a system-level feature that could add value to lots of other features. I could also see us addressing this specific task with sharing perma-links. 

Worth noting that some insights have multiple result sets. A comment might benefit from some kind of marker / annotation that points to a specific area of the insight. Likewise, insights are rarely using fixed time frames, so comments might need to point a fixed point in time to be useful. 

We touched on some of this during the collaboration work a while ago. The feedback was limited, but generally folks we spoke to wanted to push content to third party channels. https://github.com/PostHog/product-internal/issues/211
